pkg/healthcheck/healthcheck.go

```diff
@@ -1582,6 +1595,32 @@ func (hc *HealthChecker) checkCanGet(namespace, group, version, resource string)
 	return hc.checkCanPerformAction("get", namespace, group, version, resource)
 }
 
+func (hc *HealthChecker) checkAPIService(serviceName string) error {
+	apiServiceClient, err := apiregistrationv1client.NewForConfig(hc.kubeAPI.Config)
+	if err != nil {
+		return err
+	}
+
+	apiStatus, err := apiServiceClient.APIServices().Get(serviceName, metav1.GetOptions{})
+	if err != nil && len(apiStatus.Status.Conditions) == 0 {
+		return err
+	}
+
+	available := false
+	var errorMessage string
+	for _, condition := range apiStatus.Status.Conditions {
+		if condition.Type == "Available" && condition.Status == "True" {
+			available = true
+			break
+		}
+		errorMessage = fmt.Sprintf("%s: %s", condition.Reason, condition.Message)
```

Comment:
Should we restructure the above `if` statement so that we only set this errorMessage if we find an "Available" condition with status not equal to "True"?  I don't think we want to set this message if we find conditions of other types.